apps-rendering/src/image/srcsets.ts

```diff
@@ -27,7 +27,26 @@ interface Srcsets {
 
 // ----- Functions ----- //
 
-const getSubdomain = (domain: string): string => domain.split('.')[0];
+/**
+ * Get the resizing source for a given CAPI asset
+ *
+ * @param url The asset URL from CAPI
+ * @returns An image resizer source
+ */
+const getSource = (url: URL): string => {
```

Comment:
```suggestion
const getSource = (url: URL) => {
```

I think if you omit the return type, TypeScript will be able to narrow it further for you!

Reply:
You're not allowed to omit function return types in AR 🙂:

https://github.com/guardian/dotcom-rendering/blob/e9b3b1bccd75b2385efd3000dcef533e44c7eee8/apps-rendering/config/.eslintrc.js#L87

Reply:
Ah, nice one, so you would have to be explicit.

```suggestion
const getSource = (url: URL): 'static' | 'uploads' | 'media' => {
```

I may be enclined to used an `as const` object in this instance to let the compiler do its thing.